WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift

```diff
@@ -29,25 +34,74 @@ import Foundation
         WPStyleGuide.applyReaderSiteStreamCountStyle(followCountLabel)
     }
 
+    public override func sizeThatFits(size: CGSize) -> CGSize {
+        // Vertical and horizontal margins
+        let hMargin = descriptionLabel.frame.minX
+        let vMargin = descriptionLabel.frame.minY
+
+        let width = UIDevice.isPad() ? contentIPadWidthConstraint!.constant : size.width
+        let innerWidth = width - (hMargin * 2)
+        let adjustedSize = CGSize(width:innerWidth, height:CGFloat.max)
+
+        var height = UIDevice.isPad() ? contentIPadTopConstraint!.constant : 0
+        height += descriptionView.frame.minY
+        height += vMargin
+        height += descriptionLabel.sizeThatFits(adjustedSize).height
+        height += descriptionBottomConstraint.constant
+        height += followCountLabel.sizeThatFits(adjustedSize).height
+        height += followCountBottomConstraint.constant;
+
+        return CGSize(width: size.width, height: height)
+    }
+
 
    // MARK: - Configuration
 
     public func configureHeader(topic: ReaderAbstractTopic) {
-        // TODO: Wire up actual display when supported in core data
-        avatarImageView.setImageWithURL(NSURL(), placeholderImage: UIImage(named: defaultBlavatar))
-        titleLabel.text = topic.title
-        detailLabel.text = "site.com"
-        if topic.following {
+        assert(topic.isKindOfClass(ReaderSiteTopic), "Topic must be a site topic")
+
+        let siteTopic = topic as! ReaderSiteTopic
+
+        let url = NSURL(string: siteTopic.siteBlavatar)!
```

Comment:
I wasn't worried about a nil URL since the placeholder would be assigned in those cases anyway (tho a nil string would cause an exception creating the URL).  However, I see that AFNetworking has flagged the argument as non nil so let's check for that too.